src/ua.c

```diff
@@ -493,23 +494,36 @@ static bool require_handler(const struct sip_hdr *hdr,
 }
 
 
-static int sdp_af_hint(struct mbuf *mb)
+static int sdp_connection(struct mbuf *mb, int *af, struct sa *sa)
 {
-	struct pl af;
+	struct pl pl1, pl2;
+	char *addr = NULL;
 	int err;
 
+	*af = AF_UNSPEC;
 	err = re_regex((char *)mbuf_buf(mb), mbuf_get_left(mb),
-		       "IN IP[46]+", &af);
+		       "IN IP[46]+ [^ \r\n]+", &pl1, &pl2);
 	if (err)
-		return AF_UNSPEC;
+		return EINVAL;
 
-	switch (af.p[0]) {
+	pl_strdup(&addr, &pl2);
+	switch (pl1.p[0]) {
 
-	case '4': return AF_INET;
-	case '6': return AF_INET6;
+	case '4': *af = AF_INET;
+		  break;
+	case '6': *af = AF_INET6;
+		  break;
 	}
 
-	return AF_UNSPEC;
+	/* OSX/iOS needs a port number for udp_connect() */
+	err = re_regex((char *)mbuf_buf(mb), mbuf_get_left(mb),
+		       "m=audio [0-9]+ ", &pl1);
```

Comment:
not sure about this ? what if the SDP only has a video line ?

is it possible to use the re sdp module to fetch the already parsed fields ?

Reply:
Audio should be preferred. If no audio then the video line.

I will check this and extend the parsing.

Reply:
Tried to remove `sdp_connection()` by a call to `sdp_decode()` in `call_alloc()`. The result is
```
call: no common audio codecs - rejected
```
Have to postpone this. We can do this in a separate PR.